libraries/aws_s3_buckets.rb

```diff
@@ -24,15 +24,37 @@ def initialize(opts = {})
     @table = fetch_data
   end
 
+  # FilterTable setup
+  filter_table_config = FilterTable.create
+  filter_table_config.add(:bucket_names, field: :bucket_name)
+  filter_table_config.add(:tags, field: :tags)
+  filter_table_config.connect(self, :fetch_data)
+
   def fetch_data
     bucket_rows = []
     catch_aws_errors do
-      @api_response = @aws.storage_client.list_buckets.each do |resp|
-        resp.buckets.each do |bucket|
-          bucket_rows += [{ bucket_name: bucket[:name] }]
-        end
+      @api_response = @aws.storage_client.list_buckets
```

Comment:
`api_response` can be an instance variable here. Move the assignments below into the `catch_aws_errors` block.